app/src/main/java/io/gnosis/safe/helpers/ConnectivityInfoProvider.kt

```diff
@@ -18,7 +19,19 @@ class ConnectivityInfoProvider(private val connectivityManager: ConnectivityMana
 
         override fun onLost(network: Network?) {
             super.onLost(network)
-            offline = connectivityManager.activeNetworkInfo == null
+            offline = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                val capabilities =  connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+                capabilities?.let {
+                    when {
+                        capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) -> false
+                        capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) -> false
+                        else -> true
+                    }
+                } ?: true
+            } else {
+                val activeNetworkInfo = connectivityManager.activeNetworkInfo
+                activeNetworkInfo == null || !activeNetworkInfo.isConnectedOrConnecting
```

Comment:
So if the phone is currently trying to reestablish the connection `isConnectedOrConnecting` the were not offline?
Should we consider "connecting" online?

Reply:
yes, we can be more strict in this regard